GitExtUtils/GitUI/CancellationTokenSequence.cs

```diff
@@ -77,7 +88,7 @@ public CancellationToken Next()
             if (prior is null)
             {
                 next.Dispose();
-                throw new ObjectDisposedException(nameof(CancellationTokenSequence));
+                _preCancelledToken.ThrowIfCancellationRequested();
```

Comment:
💡 Either of the following also works:

```csharp
next.Cancel();
next.Dispose();
nextToken.ThrowIfCancellationRequested();
```

```csharp
next.Dispose();
new CancellationToken(true).ThrowIfCancellationRequested();
```

Reply:
I was considering the first option, but thought it would be better to have a precancelled token to not have to call `Cancel` each time. Given that it should not be a common case where Next() is extensively called for a disposed cts it should not be a problem to go for the first option you proposed.

Reply:
The performance really won't matter either way because there won't be any listeners registered for the cancellation event.